### ISSUE TITLE
[HowTo] Python

### ISSUE THREAD

**Issue by YugeTen** [YugeTen/papers-log#39]
# Numpy
To numpy array:
```python
x = np.array(x)
```
max/min along axes:
```python
np.max(x,axis=(1,2))
```
Shape
```python
x.shape
```
Concatenate in a loop (multi-dim)
```python
result_array = np.empty((0, 100))
for line in data_array:
    result = do_stuff(line) # shape: (20,100)
    result_array = np.append(result_array, [result], axis=0)
```
Concatenate in a loop (single dim)
```python
c = np.array([])
c = np.append(c,b)
```
# Random
Random value in a given shape:
```python
x = np.random.rand(dim0, dim1, dim2...)
```
Random series within range (0, 1):
```python
x = np.random.random([dim0, dim1, dim2...])
```
Random normal distribution series:
```python
x = np.random.randn(dim0, dim1, dim2...)
```
Random binary series
```python
x = np.array(np.random.choice(2, total_series_length, p = [0.5,0.5]))
```
Random integer
```python
label_index = np.random.randint(low=low_range, high=high_range)
```

# String
String formatting:
- string:
```python
# string
'{} {}'.format('one', 'two')
'{:^10}'.format('test') # 10: blank pads, ^: center

# integer
'{:d}'.format(42)
'{:4d}'.format(42) # output: 0042

# float
'{:6.2f}.format(3.1415) # output: __3.14
```
Strip digit in string:********************
```python
name_without_digit = ''.joint(str(x) for x in name if not x.isdigit())
```
Sort list X based on Y
```python
Z = [x for _,x in sorted(zip(Y,X))]
```

# Misc (list, dictionary)
Shuffle 2 lists of same length together
```python
from sklearn.utils import shuffle
list1, list2= shuffle(list1, list2)
```
Sort list X base on Y:
```python
Z = [x for _,x in sorted(zip(Y,X))]
```
Append dictionary
```python
fd.update({key:item})
```
